sycl/include/sycl/ext/oneapi/matrix/matrix-intel.hpp

```diff
@@ -557,6 +557,312 @@ inline __SYCL_ALWAYS_INLINE void joint_matrix_apply(
 #endif
 }
 
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+// Begin out-of-bounds API
+
+template <typename Group, typename T, size_t NumRows, size_t NumCols, use Use,
+          layout Layout, typename T2>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_fill_checked(
+    Group, joint_matrix<Group, T, Use, NumRows, NumCols, Layout> &res,
+    const T2 &v, size_t Stride, size_t Height, size_t Width, size_t coordX,
+    size_t coordY) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  using storage_element_type =
+      typename oneapi::detail::jm_type_interpretation_helper_trait<
+          T>::storage_element_type;
+  res.spvm = __spirv_CompositeConstructCheckedINTEL<
+      storage_element_type, T, NumRows, NumCols,
+      spv_matrix_use_traits<Use>::value,
+      spv_matrix_layout_traits<Layout>::value>(
+      static_cast<storage_element_type>(v), Stride, Height, Width, coordX,
+      coordY);
+#else
+  std::ignore = res;
+  std::ignore = v;
+  std::ignore = Stride;
+  std::ignore = Height;
+  std::ignore = Width;
+  std::ignore = coordX;
+  std::ignore = coordY;
+  throw runtime_error("joint matrix is not supported on host device.",
+                      PI_ERROR_INVALID_DEVICE);
+#endif // defined(__SYCL_DEVICE_ONLY__)
+}
+
+template <
+    typename Group, typename S, typename T, size_t NumRows, size_t NumCols,
+    access::address_space Space, access::decorated IsDecorated,
+    std::enable_if_t<std::is_same<S, std::remove_const_t<T>>::value, bool> =
+        true>
+inline __SYCL_ALWAYS_INLINE void joint_matrix_load_checked(
+    Group sg,
+    joint_matrix<Group, S, use::accumulator, NumRows, NumCols, layout::dynamic>
+        &res,
+    multi_ptr<T, Space, IsDecorated> src, size_t stride, layout Layout,
+    size_t Height, size_t Width, size_t coordX, size_t coordY) {
+#if defined(__SYCL_DEVICE_ONLY__)
+  static_assert(Space != access::address_space::private_space,
+                "Joint Matrix doesn't support load from private memory!");
+  std::ignore = sg;
+  using DecorT = typename sycl::detail::DecoratedType<T, Space>::type;
+  DecorT *Ptr = sycl::detail::getDecorated<DecorT>(src);
+  res.spvm = __spirv_JointMatrixLoadCheckedINTEL<
+      DecorT, S, NumRows, NumCols,
+      spv_matrix_use_traits<use::accumulator>::value,
+      spv_matrix_layout_traits<layout::dynamic>::value>(
+      Ptr, stride, Height, Width, coordX, coordY,
+      joint_matrix_layout_to_spv(Layout), spv_scope_traits<Group>::value);
```

Comment:
should we add detail namespace to fix test fails?
```suggestion
      detail::joint_matrix_layout_to_spv(Layout), spv_scope_traits<Group>::value);
```